Tests for REST controllers fail when the size of the test data set exceeds the page request size in PageRequestVM

**Describe the bug**
Tests for REST controllers fail when the size of the test data set exceeds the page request size in PageRequestVM when pagination is implemented.

The tests fail because they use the size of the array returned by the `.get("/api/<%= entityApiUrl %>")` clause as a proxy for the size of the dataset in the database. 

This is not correct because the size of the array is limited to the page request size in PageRequestVM if pagination is implemented. 

If the number of test data items exceeds the page request size (default is 50), the tests that use the array fail because the size is either larger than expected, or the entity cannot be found because it is outside the page that is returned.

**To Reproduce**
1.  Create a test dataset that is larger than the page request size in PageRequestVM (by default this is 50, so create a dataset with more than 50 entries)
2.  Run the REST controller test on the entity
3.  The update, create and delete tests fail

**Expected behavior**
All tests should pass irrespective of the test dataset size.

**Screenshots**
For example, the REST test for delete failes with:
```
java.lang.AssertionError: 
Expected size: 49 but was: 50 in:
[Gender{id=1, code='Toys Viaduct proactive', description='Incredible'},
    Gender{id=2, code='Cotton', description='Communications Tasty'},
    Gender{id=3, code='Kids', description='haptic generation'},
    Gender{id=4, code='sensor Account', description='Sausages Home'},
```

**Desktop (please complete the following information):**
Not relevant

**Smartphone (please complete the following information):**
Not relevant

**Additional context**
None

--- a/.github/workflows/smoke-test-imperative.yml
+++ b/.github/workflows/smoke-test-imperative.yml
@@ -152,7 +152,10 @@ jobs:
               run: npm run ci:e2e:package
             - name: 'E2E: Start docker-compose containers for e2e tests'
               if: steps.compare.outputs.equals != 'true'
-              run: npm run ci:e2e:prepare
+              run: |
+                  # Required until https://github.com/jhipster/generator-jhipster/pull/30910 is released
+                  sed -i 's/docker.io\/bitnami\/consul/hashicorp\/consul/g' src/main/docker/consul.yml || true
+                  npm run ci:e2e:prepare
             - name: 'E2E: Run'
               if: steps.compare.outputs.equals != 'true'
               id: e2e

--- a/generators/quarkus/templates/partials/data_application.properties.ejs
+++ b/generators/quarkus/templates/partials/data_application.properties.ejs
@@ -99,6 +99,7 @@ quarkus.mongodb.database=<%= baseName %>
 <%_ if (databaseType === 'sql') { _%>
 quarkus.liquibase.change-log=config/liquibase/master.xml
 quarkus.liquibase.migrate-at-start=true
+%test.quarkus.liquibase.contexts=test
 <%_ if (prodDatabaseType !== devDatabaseType) { _%>
 %test.quarkus.datasource.jdbc.url=jdbc:h2:mem:test;MODE=LEGACY
 %test.quarkus.datasource.db-kind=h2

